src/chooser_factory.cpp

```diff
@@ -0,0 +1,31 @@
+#include "chooser_factory.h"
+#include "setting_option.h"
+#include "random_chooser.h"
+#include "ml_chooser.h"
+
+#include <iostream>
+
+using namespace std;
+using namespace myrps;
+
+Chooser *ChooserFactory::MakeChooser(SettingOption choice)
+{
+  if(choice == SettingOption::D_RANDOM)
+  {
+    return new RandomChooser();
+  }
+  else if(choice == SettingOption::D_SMART)
+  {
+    return new MLChooser();
```

Comment:
Will likely change after merging #34, but that can be resolved in a later issue.

Reply:
will resolve in #36